packages/inference/src/providers/providerHelper.ts

```diff
@@ -0,0 +1,137 @@
+import type { ChatCompletionOutput, TextGenerationOutput } from "@huggingface/tasks";
+import { InferenceOutputError } from "../lib/InferenceOutputError";
+import type { BodyParams, HeaderParams, UrlParams } from "../types";
+import { toArray } from "../utils/toArray";
+/**
+ * Base class for task-specific provider helpers
+ */
+export abstract class TaskProviderHelper {
```

Comment:
(nit) let's group all `asbtract` functions together?
I think it would improve readability

```typescript
export abstract class TaskProviderHelper {
	/// [...]
	abstract getResponse(/** [...] */): unknown;
	abstract makeRoute(/** [...] */): unknown;
	abstract prepareBody(/** [...] */): unknown;
	
	makeBaseUrl(): string {
		return this.baseUrl;
	}
	/// [...]
}
```